x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/ApiKeyService.java

```diff
@@ -360,6 +362,27 @@ && hasRemoteIndices(request.getRoleDescriptors())) {
         }
     }
 
+    private Set<RoleDescriptor> removeUserRoleDescriptorDescriptions(Set<RoleDescriptor> userRoleDescriptors) {
```

Comment:
API key's role descriptors do not support description field. Owner's role on the other hand may contain description. Not removing description from user's (limited-by) role descriptors would cause BWC issue. I think it is simpler and consistent to not store role description at all.